lampod/src/liquidity.rs

```diff
@@ -169,28 +166,48 @@ impl LampoLiquidityManager {
     // Select the best fee_param from a list of fee_param given by the lsp provider
     // and then forward the request to the LSP for invoice generation
     // This will respond in InvoiceParametersReady event
-    pub fn buy_request(
-        &self,
-        best_fee_param: OpeningFeeParams,
-        amount_msat: u64,
-    ) -> error::Result<()> {
+    fn buy_request(&self, best_fee_param: OpeningFeeParams, amount_msat: u64) -> error::Result<()> {
         let node_id = self.lsp_provider.as_ref().unwrap().node_id;
-        let res = self
-            .lampo_liquidity
+        self.lampo_liquidity
             .lsps2_client_handler()
             .unwrap()
-            .select_opening_params(node_id, Some(amount_msat), best_fee_param);
+            .select_opening_params(node_id, Some(amount_msat), best_fee_param)
+            .map_err(|err| error::anyhow!("Error Occured : {:?}", err))?;
 
         Ok(())
     }
 
-    pub fn create_a_jit_channel(&self) {
-        todo!()
+    pub fn create_a_jit_channel(
+        &self,
+        amount_msat: u64,
+        description: String,
+    ) -> error::Result<Bolt11Invoice> {
+        self.client_request_opening_params()?;
+        // TODO: We need to provide a suitable algorithm to get the best_params from all the
+        // opening params that we get from the peer. For now we are getting the first param
+        let best_fee_param = self
+            .lsp_provider
+            .as_ref()
+            .unwrap()
+            .opening_params
+            .as_ref()
+            .unwrap()[0]
+            .clone();
```

Comment:
when you will stop chaining 300 method? This is bad for readiblity

Reply:
plus `.opening_params
            .as_ref()
            .unwrap()[0]` looks wrong